examples/guide/examples/guide.rs

```diff
@@ -1,12 +1,13 @@
+use burn::backend::{WgpuAutodiffBackend, WgpuBackend};
+use burn::data::dataset::Dataset;
 use burn::optim::AdamConfig;
-use burn_dataset::Dataset;
 use guide::{model::ModelConfig, training::TrainingConfig};
 
 fn main() {
-    type MyBackend = burn_wgpu::WgpuBackend<burn_wgpu::AutoGraphicsApi, f32, i32>;
-    type MyAutodiffBackend = burn_autodiff::ADBackendDecorator<MyBackend>;
+    type MyBackend = WgpuBackend;
+    type MyAutodiffBackend = WgpuAutodiffBackend;
```

Comment:
For this example, let's keep the generics. It's documented this way in the guide and shows how to set the element types! But having defaults is indeed convenient.